storage.go

```diff
@@ -83,11 +84,16 @@ func (f *fsStorage) Save(key, etag string, data []byte) error {
 	}
 
 	filePath := f.filePath(key)
+	filePathTmp := filePath + tmpSuffix
 
 	// Creates the file if it doesn't exist, or truncates it if it does.
-	file, err := os.Create(filePath)
+	// We use a temporary file to reduce the chance of corrupted files due to having to stop midair.
+	// We do not have to check for remnants from a previous run because if there is a left-over temporary file, we will
+	// overwrite it with the next run and rename it afterward.
+	// Therefore, left-overs will be gone after the next run.
+	file, err := os.Create(filePathTmp)
```

Comment:
Why not use `os.CreateTemp`?

Reply:
Good point.
But I had two arguments for picking the other approach:
- When stopping midair we will end up half-baked files; when always using the same file path things cannot pile up and a successful run will clear all traces of the unsuccessful ones. When using `os.CreateTemp` that wouldn't work - though, stuff would get cleaned up when restarting.
- It makes checking for such remnants easier because one knows where to look for them

Additionally, but I could not find actual proof for that theory and probably it does not really matter: I expect it to be cheaper to rename a file within a directory than across folders due to requiring less fiddling with the filesystem's btrees and stuff. 😉 

Does that convince you? ;)

Reply:
I just realised: `/tmp` might be another disk/filesystem - so in this case "renaming" would be a proper, "expensive" move.